django/core/checks/model_checks.py

```diff
@@ -64,8 +64,10 @@ def extract_operation(obj):
         """
         operation, args, keywords = obj, [], {}
         while hasattr(operation, 'func'):
-            args.extend(getattr(operation, 'args', []))
-            keywords.update(getattr(operation, 'keywords', {}))
+            # The or clauses here are redundant but work around a
+            # bug in functools.partial in certain Python versions.
```

Comment:
Yep wasn't thinking long-term enough. I've specified the exact versions and put the Python issue number in the comment.